### ISSUE TITLE
Purchasing a new skill throws Primary key violation inserting new skill row

### ISSUE THREAD

**Issue by spazmodica** [ACEmulator/ACE#635]
### Bug Report or Feature Request (mark with an `x`)
```
- [X] bug report -> please search issues before submitting
- [ ] feature request
```

### Repro steps.
1. Create character
2. Grant experience enough to get enough skill points to purchase a new skill
3. Spend skill points on new skill
4. Wait a bit
5. Server throws exception

### The log given by the failure.
{MySql.Data.MySqlClient.MySqlException (0x80004005): Duplicate entry '1342177282-23' for key 'PRIMARY'
   at MySql.Data.MySqlClient.MySqlStream.ReadPacket()
   at MySql.Data.MySqlClient.NativeDriver.GetResult(Int32& affectedRow, Int64& insertedId)
   at MySql.Data.MySqlClient.Driver.NextResult(Int32 statementId, Boolean force)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResult()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader(CommandBehavior behavior)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at ACE.Database.Database.DatabaseTransaction.<>c__DisplayClass10_0.<Commit>b__0() in D:\Repos\spazmodica\ACE\Source\ACE.Database\Database.cs:line 200
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)}	MySql.Data.MySqlClient.MySqlException

### Desired functionality.
Purchasing new skill should not throw error.

### Mention any other details that might be useful.
Exception is in Task so does not crash server
Inserts vs Updates are determined by the IsDirty and HasEverBeenSavedToDatabase flags
Skills are all added to db upon creation, but purchasing does not set above flags correctly triggering error

**Comment by Mag-nus** [comment]
I don't think this bug exists in the new EF framework.

We'll need to re-test when that gets pushed to master.

**Comment by LtRipley36706** [comment]
I don't see this on EF branch either currently

**Comment by LtRipley36706** [comment]
Fixed in #675
